features/other_b/skin.feature

```diff
@@ -433,3 +433,19 @@ Feature: creating and editing skins
     And I should see "only screen and (max-width: 42em), only screen and (max-width: 62em)"
   When I press "Use"
   Then the page should have a skin with the media query "only screen and (max-width: 42em), only screen and (max-width: 62em)"
+
+  Scenario: A user should be able to choose a temporary skin
+  Given basic skins
+    And the approved public skin "public skin" with css "#title { text-decoration: blink;}"
+   Then I am logged in as an admin
```

Comment:
Is the reason we're doing these log in as admin/check "Cache"/etc steps to make sure the skin is cached so we can put it in the chooser? If so, I'd suggest setting it up as a step, like you did with "the skin  "([^\"]*)" is in the chooser":
```
When /^the skin "([^\"]*)" is cached$/ do |skin_name|
  skin = Skin.find_by_title(skin_name)
  skin.cached = true
  skin.save
  skin.cache!
end
```

I might also add two steps to make sure we're actually using (or not using) the cached skin rather than relying solely on the success message:
```
Then /^the page should have the cached skin "([^"]*)"$/ do |skin_name|
  skin = Skin.find_by_title(skin_name)
  page.should have_xpath("//link[contains(@href, '#{skin.skin_dirname}')]")
end

Then /^the page should not have the cached skin "([^"]*)"$/ do |skin_name|
  skin = Skin.find_by_title(skin_name)
  page.should_not have_xpath("//link[contains(@href, '#{skin.skin_dirname}')]")
end
```

Then I would do:
```
  Scenario: User can set a skin for a session and then unset it
  Given basic skins
    And the approved public skin "public skin" with css "#title { text-decoration: blink;}"
    And the skin "public skin" is cached
    And the skin "public skin" is in the chooser
  When I am logged in as "skinner"
    And I follow "public skin"
  Then I should see "The skin public skin has been set. This will last for your current session."
    And the page should have the cached skin "public skin"
  When I follow "Default"
  Then I should see "You are now using the default Archive skin again!"
    And the page should not have the cached skin "public skin"
```